static/gsApp/components/quotaExceededNavItem.tsx

```diff
@@ -0,0 +1,292 @@
+      </Header>
+      <Body>
+        <Title>{t('Quota Exceeded')}</Title>
+        <Description>
+          {tct(
+            'You’ve run out of [exceededCategories] for this billing cycle. This means we are no longer monitoring or ingesting events and showing them in Sentry.',
+            {
+              exceededCategories: listDisplayNames({
+                plan: subscription.planDetails,
+                categories: exceededCategories,
+                hadCustomDynamicSampling: subscription.hadCustomDynamicSampling,
+              }),
+            }
+          )}
+        </Description>
+        <ActionContainer>
+          <AddEventsCTA
+            organization={organization}
+            subscription={subscription}
+            buttonProps={{
+              size: 'xs',
+            }}
+            eventTypes={eventTypes}
+            notificationType="overage_critical"
+            referrer={`overage-alert-sidebar-${eventTypes.join('-')}`}
+            source="quota-overage"
+            handleRequestSent={() => {}}
+          />
+          <DismissContainer>
+            <Checkbox
+              name="dismiss"
+              checked={isDismissed}
+              disabled={isDismissed}
+              onChange={() => {
+                onDismiss();
+              }}
+            />
+            <CheckboxLabel>{t("Don't annoy me again")}</CheckboxLabel>
+          </DismissContainer>
+        </ActionContainer>
+      </Body>
+    </Container>
+  );
+}
+
+function PrimaryNavigationQuotaExceeded({
+  subscription,
+  organization,
+}: {
+  organization: Organization;
+  subscription: Subscription;
+}) {
+  const [exceededCategories, setExceededCategories] = useState<string[]>([]);
+  const [promptsToCheck, setPromptsToCheck] = useState<string[]>([]);
+  const [isDismissed, setIsDismissed] = useState(false);
+  const [hasLoaded, setHasLoaded] = useState(false);
+
+  useEffect(() => {
+    if (!subscription.categories) {
+      return;
+    }
+    const result = sortCategoriesWithKeys(subscription?.categories ?? {})
+      .filter(
+        ([category]) =>
+          category !== DataCategory.SPANS_INDEXED ||
+          subscription?.hadCustomDynamicSampling
+      )
+      .reduce((acc, [category, currentHistory]) => {
+        if (currentHistory.usageExceeded) {
+          acc.push(category);
+        }
+        return acc;
+      }, [] as string[]);
+    setExceededCategories(result);
+    setPromptsToCheck(
+      result
+        .map(category => {
+          const categoryInfo = getCategoryInfoFromPlural(category as DataCategory);
+          if (categoryInfo) {
+            return `${categoryInfo.snakeCasePlural ?? category}_overage_alert`;
+          }
+          return null;
+        })
+        .filter(Boolean) as string[]
+    );
+  }, [subscription.categories, subscription.hadCustomDynamicSampling]);
+
+  const {
+    data: prompts,
+    isPending,
+    isError,
+  } = usePromptsCheck(
```

Comment:
Can you use `usePrompt()` instead? That should simplify a lot of the dismissal logic here

Reply:
Since we're checking and updating multiple prompts, I introduced a new hook that is basically a copy-paste of `usePrompt()` but with multiple strings. Let me know what you think or if it'd be more ideal to rework `usePrompt()` instead of adding an entirely new hook!